### PR TITLE
Add -Werror mode and use it for testing

### LINKED ISSUE #417: Compiler warnings
I document some compiler warnings here:
```
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/xsimd_avx512_conversion.hpp: In function ‘xsimd::batch<double, 8> xsimd::bitwise_cast(const xsimd::batch_bool<double, 8>&)’:
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/../types/xsimd_base.hpp:606:10: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         *reinterpret_cast<int_type*>(&z1) = ~int_type(0);                      \
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/../types/xsimd_base.hpp:611:5: note: in expansion of macro ‘XSIMD_DEFINE_BITWISE_CAST_FLOAT’
     XSIMD_DEFINE_BITWISE_CAST_FLOAT(double, NMIN)                              \
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/xsimd_avx512_conversion.hpp:47:5: note: in expansion of macro ‘XSIMD_DEFINE_BITWISE_CAST_ALL’
     XSIMD_DEFINE_BITWISE_CAST_ALL(8)
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/xsimd_avx512_conversion.hpp: In function ‘xsimd::batch<float, 16> xsimd::bitwise_cast(const xsimd::batch_bool<float, 16>&)’:
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/../types/xsimd_base.hpp:606:10: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         *reinterpret_cast<int_type*>(&z1) = ~int_type(0);                      \
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/../types/xsimd_base.hpp:612:5: note: in expansion of macro ‘XSIMD_DEFINE_BITWISE_CAST_FLOAT’
     XSIMD_DEFINE_BITWISE_CAST_FLOAT(float, NMIN * 2)                           \
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/tdegeus/miniconda3/envs/code_velocity_s6g1/include/xsimd/types/xsimd_avx512_conversion.hpp:47:5: note: in expansion of macro ‘XSIMD_DEFINE_BITWISE_CAST_ALL’
     XSIMD_DEFINE_BITWISE_CAST_ALL(8)
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Fix #417

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
@JohanMabille this one is ready for review!

File: include/xsimd/types/xsimd_sse_double.hpp [hunk 1]
```diff
@@ -633,6 +633,7 @@ namespace xsimd
             static batch_type select(const batch_bool_constant<value_type, Values...>& cond, const batch_type& a, const batch_type& b)
             {
 #if XSIMD_X86_INSTR_SET >= XSIMD_X86_SSE4_1_VERSION
+                (void)cond;
```
Comment (JohanMabille) [inline include/xsimd/types/xsimd_sse_double.hpp:636]: Is it possible to remove `cond` from the signature instead? (like you did in `xsimd_avx_float`)

**Comment by serge-sans-paille** [comment]
Not really because of the else part of the #ifdef

**Comment by JohanMabille** [comment]
Ha indeed, I didn't see it in the diff since it has not changed.

**Pull request merged**